oasislmf/execution/bash.py

```diff
@@ -1075,94 +1150,90 @@ def genbash(
+    bash_params['fmpy_low_memory'] = fmpy_low_memory
+    bash_params['fmpy_sort_output'] = fmpy_sort_output
+    bash_params['process_number'] = process_number
+    bash_params['remove_working_files'] = remove_working_files
+    bash_params['model_run_dir'] = model_run_dir
+    bash_params['gul_threshold'] = analysis_settings.get('gul_threshold', 0)
+    bash_params['number_of_samples'] = analysis_settings.get('number_of_samples', 0)
+
+    # set complex model gulcalc command
+    if not _get_getmodel_cmd and custom_gulcalc_cmd:
+        bash_params['_get_getmodel_cmd'] = get_complex_model_cmd(custom_gulcalc_cmd, analysis_settings)
+    else:
+        bash_params['_get_getmodel_cmd'] = _get_getmodel_cmd
 
-    use_random_number_file = False
-    stderr_guard = stderr_guard
-    gul_item_stream = not gul_legacy_stream
-    full_correlation = False
-    fifo_queue_dir = ""
-    fifo_full_correlation_dir = ""
-    work_dir = 'work/'
-    work_kat_dir = 'work/kat/'
-    work_full_correlation_dir = 'work/full_correlation/'
-    work_full_correlation_kat_dir = 'work/full_correlation/kat/'
-    output_dir = 'output/'
-    output_full_correlation_dir = 'output/full_correlation/'
+    ## Set fifo dirs
+    if fifo_tmp_dir:
+        bash_params['fifo_queue_dir'] = '/tmp/{}/fifo/'.format(''.join(random.choice(string.ascii_letters + string.digits) for _ in range(10)))
+    else:
+        bash_params['fifo_queue_dir'] = os.path.join(model_run_dir, 'fifo/')
+
+    ## set dirs
+    bash_params['stderr_guard'] = stderr_guard
+    bash_params['gul_item_stream'] = not gul_legacy_stream
+    bash_params['work_dir'] = os.path.join(model_run_dir, 'work/')
+    bash_params['work_kat_dir'] = os.path.join(model_run_dir, 'work/kat/')
+    bash_params['work_full_correlation_dir'] = os.path.join(model_run_dir, 'work/full_correlation/')
+    bash_params['work_full_correlation_kat_dir'] = os.path.join(model_run_dir, 'work/full_correlation/kat/')
+    bash_params['output_dir'] = os.path.join(model_run_dir, 'output/')
+    bash_params['output_full_correlation_dir'] = os.path.join(model_run_dir, 'output/full_correlation/')
+    bash_params['fifo_full_correlation_dir'] = os.path.join(bash_params['fifo_queue_dir'], 'full_correlation/')
 
     # Set default alloc/shuffle rules if missing
-    gul_alloc_rule = gul_alloc_rule if isinstance(gul_alloc_rule, int) else KTOOLS_ALLOC_GUL_DEFAULT
-    il_alloc_rule = il_alloc_rule if isinstance(il_alloc_rule, int) else KTOOLS_ALLOC_IL_DEFAULT
-    ri_alloc_rule = ri_alloc_rule if isinstance(ri_alloc_rule, int) else KTOOLS_ALLOC_RI_DEFAULT
-    num_gul_per_lb = num_gul_per_lb if isinstance(num_gul_per_lb, int) else KTOOL_N_GUL_PER_LB
-    num_fm_per_lb = num_fm_per_lb if isinstance(num_fm_per_lb, int) else KTOOL_N_FM_PER_LB
-    event_shuffle = event_shuffle if isinstance(event_shuffle, int) else EVE_DEFAULT_SHUFFLE
+    bash_params['gul_alloc_rule'] = gul_alloc_rule if isinstance(gul_alloc_rule, int) else KTOOLS_ALLOC_GUL_DEFAULT
```

Comment:
wouldn't this be better just having a default dictionary that we override if there is an instance for instance:

```python 
class BashParams(dict):

    DEFAULT = {
        "one": 1,
        . . . 
    }

    def __init__(self):
        super().__init__(self.DEFAULT)

   def add_int(self, input_value, key_value):
       if isinstance(input_value):
           self[key_value] = input_value
```
This will give us a reference point where we can see what are the defaults in one point, and adding the parameter is less code as there is a lot of checks here

Reply:
I agree, thats a better approach.  

I'll merge this PR to get the release candidate rolling and open a new issue to update the `bash_params` using your comment